maestro-cli/src/main/java/maestro/cli/cloud/CloudInteractor.kt

```diff
@@ -328,42 +348,34 @@ class CloudInteractor(
     private fun saveReport(
         reportFormat: ReportFormat,
         passed: Boolean,
-        upload: UploadStatus,
+        suiteResult: TestExecutionSummary.SuiteResult,
         reportOutputSink: BufferedSink,
         testSuiteName: String?
     ) {
         ReporterFactory.buildReporter(reportFormat, testSuiteName)
             .report(
                 TestExecutionSummary(
                     passed = passed,
-                    suites = listOf(
-                        TestExecutionSummary.SuiteResult(
-                            passed = passed,
-                            flows = upload.flows.map { flow ->
-                                val failure = flow.errors.firstOrNull()
-                                TestExecutionSummary.FlowResult(
-                                    name = flow.name,
-                                    fileName = null,
-                                    status = FlowStatus.from(flow.status),
-                                    failure = if (failure != null) TestExecutionSummary.Failure(failure) else null
-                                )
-                            }
-                        )
-                    )
+                    suites = listOf(suiteResult)
                 ),
                 reportOutputSink,
             )
     }
 
-    companion object {
-
-        private val COMPLETED_STATUSES = setOf(
-            UploadStatus.Status.CANCELED,
-            UploadStatus.Status.WARNING,
-            UploadStatus.Status.SUCCESS,
-            UploadStatus.Status.ERROR,
+    private fun convert(passed: Boolean, upload: UploadStatus, runningFlows: RunningFlows): TestExecutionSummary.SuiteResult {
```

Comment:
let's be more specific about what this method converts by giving it a better name

Reply:
applied